gtest/src/manager.rs

```diff
@@ -278,23 +265,31 @@ impl ExtManager {
     }
 
     fn init_success(&mut self, message_id: MessageId, program_id: ProgramId) {
-        let (_, state, _) = self
+        let (actor, _) = self
             .actors
             .get_mut(&program_id)
             .expect("Can't find existing program");
 
-        *state = ProgramState::Initialized;
+        if let Actor::Uninitialized(_, maybe_prog) = actor {
+            assert!(maybe_prog.is_some(), "there is always program by contract");
+            if let Some(Program::Genuine(p)) = maybe_prog {
+                p.set_initialized();
+            }
+            Actor::transform_to_initialized(actor);
```

Comment:
Why such a kind of call?

```suggestion
            actor.transform_to_initialized();
```

Reply:
I thought it is more readable. The name of the call tells us, that something is mutated (transformed). More readable version is not "self-transform`, but "transform->what->self". After changing the name to `set_initialized` it will be like `actor.set_initialized()`.

Reply:
done